src/Zynga/Framework/PgData/V1/PgModel/Reader.hh

```diff
@@ -65,52 +67,51 @@ class Reader implements ReaderInterface {
       // 1) grab a copy of our object to work with.
       $pgModel->stats()->incrementCacheMisses();
       $obj = $pgModel->data()->createRowObjectFromClassName($model);
```

Comment:
Following 3 lines of code not used. We just using this object to get PK for where condition.
fetchResultSetFromDatabase() - internally does the similar approach to create the fresh objects for RS and apply locks on them.
am I missing anything?

Reply:
i was looking at fetchResultSetFromDatabase and i didn't see where it was grabbing a lock on the row.

Reply:
it's very bottom of the method, it's grabbing lock only if it's not existed on cache tho.
        // X) Fetch cached data to allow for delayed sync back from mc -> db
        // We dont lock on entire data sets
        $cachedRow =
          $this->fetchSingleRowFromDataCache($model, $pkValue, false);

        if ($cachedRow instanceof PgRowInterface) {
          $resultSet->add($cachedRow);
        } else {
          $pgModel->cache()->lockRowCache($obj);
          $this->setSingleRowToDataCache($obj, $releaseLockOnSet);
          $resultSet->add($obj);
        }

Reply:
Only in the else case. There can be some edge case where a second thread wrote data to data cache and that would cause fetchSingleRowFromDataCache go in the if check and not grab a lock at all.